src/main/java/org/folio/circulation/domain/OverdueFineCalculatorService.java

```diff
@@ -217,7 +219,10 @@ private OverdueFineCalculatorService(Clients clients) {
       .thenCompose(r -> r.after(this::lookupItemRelatedRecords))
       .thenCompose(r -> r.after(this::lookupFeeFineOwner))
       .thenCompose(r -> r.after(params -> this.lookupLoggedInUser(params, loggedInUserId)))
-      .thenCompose(r -> r.after(params -> createAccount(fineAmount, params)));
+      .thenCompose(r -> r.after(params -> createAccount(fineAmount, params)))
+      .thenCompose(r -> r.after(feeFineAction ->
+        repos.scheduledNoticesRepository.deleteOverdueNotices(loan.getId())
+          .thenApply(rs -> succeeded(feeFineAction))));
```

Comment:
This operation seems rewrite overall result, success is returned regardless previous results at the chain.

Reply:
Ignoring `deleteOverdueNotices()` result was my intent because deletion of these notices is not a crucial part of the overdue fine charging process. If `deleteOverdueNotices()` causes the whole process to fail, the fine will still be charged, but overdue fine notices won't be scheduled further down the chain. This means that patron won't be noticed about the fine, which I think is much worse than extra notices about overdue.

What wasn't my intent though is for this result to be explicitly succeeded, so I've replaced this now with a simple mapping to the previous result (of the `createAccount()` method). But actually, if `createAccount()` fails, the `.after()` code will not run at all, so I believe it'll always be succeeded anyway.